circuit/src/wrong/native.rs

```diff
@@ -0,0 +1,371 @@
+use super::rns::{compose, decompose_big, fe_to_big, RnsParams};
+use halo2wrong::halo2::arithmetic::FieldExt;
+use num_bigint::BigUint;
+use std::marker::PhantomData;
+
+/// Enum for the two different type of Quotient.
+#[derive(Clone, Debug)]
+enum Quotient<W: FieldExt, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P>
+where
+	P: RnsParams<W, N, NUM_LIMBS, NUM_BITS>,
+{
+	/// Quotient type for the addition.
+	Add(N),
+	/// Quotient type for the multiplication.
+	Mul(Integer<W, N, NUM_LIMBS, NUM_BITS, P>),
+}
+
+impl<W: FieldExt, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P>
+	Quotient<W, N, NUM_LIMBS, NUM_BITS, P>
+where
+	P: RnsParams<W, N, NUM_LIMBS, NUM_BITS>,
+{
+	/// Returns Quotient type for the addition.
+	pub fn add(self) -> N {
+		match self {
+			Quotient::Add(res) => res,
+			_ => Option::<N>::None.unwrap(),
```

Comment:
Why not just return Option and unwrap it when you want to use it

Reply:
Do you mean returning Some(Quotient::Add(self)).unwrap()? I thought returning a value would not be good for exception handling

Reply:
```rust
pub fn add(self) -> Option<N>
```
would still be better than calling `.unwrap()` inside this function

Reply:
This could be a solution until we implement a proper error handling.
But also I see that `quotient.add()` is never called, so it might not be needed

Reply:
Okay. Also, why coverage check is failing? It seems like a problem from server side.